config/routes.rb

```diff
@@ -0,0 +1,13 @@
+Rails.application.routes.draw do
+  root to: "works#top_ten"
+  get "/login", to: "users#login_form", as: :login
+  post "/login", to: "users#login"
+  post "/logout", to: "users#logout", as: :logout
+  patch "/works/:id/upvote", to: "votes#upvote", as: :upvote
+  resources :votes
+  resources :works
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  
+  resources :users
```

Comment:
This also creates all 7 restful routes when only index and show are actually setup in the controller.
```suggestion
  resources :users, only: [:index, :show]
```